### PR TITLE
Add progress bar to app load

### PR DESCRIPTION
**Pull request by zachgibb**

### CODE REVIEW

File: src/sagas/authentication.js [hunk 1]
```diff
@@ -187,12 +187,26 @@ export default function* authenticationWatcher() {
       const mnemonicPhrase = yield select(mnemonicPhraseSelector);
       const pin = yield call(getKey);
 
+      let to;
+      let resetReplace = false;
       if (mnemonicPhrase && pin) {
-        NavigatorService.resetReplace('Login', 'Menu');
+        resetReplace = true;
+        to = 'Menu';
```
Comment (ericgibb) [inline src/sagas/authentication.js]: can we start to use translations here:
```
import { t } from 'translations/i18n';

t('authentication.redirect_to_menu')
```

File: src/sagas/authentication.js [hunk 2]
```diff
@@ -187,12 +187,26 @@ export default function* authenticationWatcher() {
...
-        NavigatorService.navigate('Mnemonic');
+        to = 'Mnemonic';
+      }
+
+      const currentRoute = NavigatorService.getCurrentRoute().routes[0].routeName;
+      if (currentRoute === 'LoadingScreen') {
+        yield put({type: 'POST_LOAD_REDIRECT', to});
+      } else {
+        if (resetReplace) {
+          NavigatorService.resetReplace(currentRoute, to);
```
Comment (ericgibb) [inline src/sagas/authentication.js]: 👍

File: src/screens/LoadingScreen/LoadingScreen.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    const {
+      appInitialized,
+      walletHydrated,
+      transactionsHydrated,
+      pricingInitialized
+    } = this.props;
+
+    let descriptions = [];
+    if (!appInitialized) {
+      descriptions.push('Initializing App');
```
Comment (ericgibb) [inline src/screens/LoadingScreen/LoadingScreen.js]: i18n these, please

File: src/components/ProgressBar.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    });
+
+    return (
+      <View style={[style, styles.container]}>
+        <View style={styles.progressContainer} onLayout={this.handleLayout}>
+          <Animated.View style={[styles.progressAnimation, {transform: [{translateX}]}]}>
+            <LinearGradient
+              start={gradients.horizontal.start}
+              end={gradients.horizontal.end}
+              colors={[ '#E5129A', '#9E0C6A' ]}
```
[the remark this answers is not in the document]
Comment (zachgibb) [reply src/components/ProgressBar.js]: these are the same visual colors as the regular pink gradient, but they are with the updated color profile - i've updated the colors themselves and then will just use `gradients.pink` 
7bae543

File: src/sagas/authentication.js (hunk 2 above)
Comment (zachgibb) [reply src/sagas/authentication.js]: 🦄

File: src/sagas/authentication.js (hunk 1 above)
Comment (zachgibb) [reply src/sagas/authentication.js]: those are for the route names, which wouldn't change based on which translation the app is using, but i will pull in the translations to the other files for sure

File: src/screens/LoadingScreen/LoadingScreen.js (hunk 3 above)
Comment (zachgibb) [reply src/screens/LoadingScreen/LoadingScreen.js]: fixed 
a296e05

File: src/sagas/authentication.js (hunk 1 above)
Comment (ericgibb) [reply src/sagas/authentication.js]: 👍

**Pull request merged**